--- src/Makefile.am~	2017-03-07 18:35:42.000000000 +0000
+++ src/Makefile.am	2017-03-07 18:35:58.950961825 +0000
@@ -27,11 +27,11 @@
 		  prelinktab.h reloc.c reloc.h space.c undo.c undoall.c      \
 		  verify.c md5.c md5.h sha.c sha.h 	     \
 		  $(common_SOURCES) $(arch_SOURCES)
-prelink_LDADD = @LIBGELF@ -liberty
+prelink_LDADD = @LIBGELF@ -liberty -lbfd
 prelink_LDFLAGS =
 
 execstack_SOURCES = execstack.c $(common_SOURCES) $(arch_SOURCES)
-execstack_LDADD = -liberty
+execstack_LDADD = -liberty -lbfd
 execstack_LDFLAGS =
 
 extra_DIST = makecrc.c
