--- gc.h.orig	2013-08-21 04:46:10.647150548 +0000
+++ gc.h	2013-08-21 04:49:10.701436795 +0000
@@ -2,9 +2,9 @@
 #ifndef RUBY_GC_H
 #define RUBY_GC_H 1
 
-#if defined(__x86_64__) && defined(__GNUC__)
+#if defined(__x86_64__) && defined(__LP64__) && defined(__GNUC__)
 #define SET_MACHINE_STACK_END(p) __asm__ volatile ("movq\t%%rsp, %0" : "=r" (*(p)))
-#elif defined(__i386) && defined(__GNUC__)
+#elif (defined(__i386) || (defined(__x86_64__) && defined(__ILP64__))) && defined(__GNUC__)
 #define SET_MACHINE_STACK_END(p) __asm__ volatile ("movl\t%%esp, %0" : "=r" (*(p)))
 #else
 NOINLINE(void rb_gc_set_stack_end(VALUE **stack_end_p));
