--- examples/Makefile.am.orig	2013-04-05 23:47:30.805662981 +0100
+++ examples/Makefile.am	2013-04-05 23:50:02.631916035 +0100
@@ -5,16 +5,14 @@
 #     See COPYRIGHT in top-level directory.
 #
 
-# FIXME should these be prefixed with AM_ under automake?  Should we cut some or
-# all of these?
 MPICHDIR=..
 #CC       = $(MPICHDIR)/bin/mpicc
-CFLAGS   = $(WRAPPER_CFLAGS)
-CXXFLAGS = $(WRAPPER_CXXFLAGS)
-FFLAGS   = $(WRAPPER_FFLAGS)
-FCFLAGS  = $(WRAPPER_FCFLAGS)
-LDFLAGS  =
-CPPFLAGS = $(WRAPPER_CPPFLAGS)
+AM_CFLAGS   = $(WRAPPER_CFLAGS)
+AM_CXXFLAGS = $(WRAPPER_CXXFLAGS)
+AM_FFLAGS   = $(WRAPPER_FFLAGS)
+AM_FCFLAGS  = $(WRAPPER_FCFLAGS)
+AM_LDFLAGS  =
+#AM_CPPFLAGS = $(WRAPPER_CPPFLAGS)
 #DEFS     = 
 
 AM_CPPFLAGS = -I$(top_builddir)/src/include -I${top_srcdir}/src/include
@@ -34,14 +32,14 @@
 # is a better system?
 
 # the make-time instances of libpmpich.la and libmpich.la live here
-LDFLAGS += -L../lib
+AM_LDFLAGS += -L../lib
 # the make-time instances of libmpl.la and libopa.la live here
-LDFLAGS += -L../src/mpl -L../src/openpa/src
+AM_LDFLAGS += -L../src/mpl -L../src/openpa/src
 
 # Wrapper LDFLAGS need to be added at the end to make sure we link
 # with the libraries we just built, and not any previously installed
 # libraries.
-LDFLAGS += $(WRAPPER_LDFLAGS)
+AM_LDFLAGS += $(WRAPPER_LDFLAGS)
 
 external_libs = -lmpl -lopa $(WRAPPER_LIBS)
 if BUILD_PROFILING_LIB
@@ -76,8 +76,8 @@
 # --disable-static" is passed to configure.
 cpi_SOURCES = cpi.c
 cpi_LDADD = -lm
-cpi_LDFLAGS = $(LDFLAGS) $(mpich_libtool_static_flag)
-cpi_CFLAGS = $(CFLAGS)
+cpi_LDFLAGS = $(AM_LDFLAGS) $(mpich_libtool_static_flag)
+cpi_CFLAGS = $(AM_CFLAGS)
 
 icpi_SOURCES = icpi.c
 icpi_LDADD = -lm
