--- libs/log/build/Jamfile.v2~	2015-04-04 18:33:00.000000000 +0100
+++ libs/log/build/Jamfile.v2	2016-02-25 22:50:32.871359855 +0000
@@ -249,7 +249,7 @@
             result = ;
             if <toolset>gcc in $(properties) || <toolset>clang in $(properties)
             {
-                result = <cxxflags>"-march=core2 -msse -msse2 -msse3 -mssse3" ;
+                result = <cxxflags>"-msse -msse2 -msse3 -mssse3" ;
             }
             else if <toolset>intel in $(properties)
             {
@@ -311,7 +311,7 @@
             result = ;
             if <toolset>gcc in $(properties)
             {
-                result = <cxxflags>"-march=core-avx2 -mavx -mavx2 -fabi-version=0" ;
+                result = <cxxflags>"-mavx -mavx2 -fabi-version=0" ;
             }
             else if <toolset>clang in $(properties)
             {
--- tools/build/src/tools/gcc.jam~	2016-02-27 23:16:45.000000000 +0000
+++ tools/build/src/tools/gcc.jam	2016-02-27 23:29:24.581019880 +0000
@@ -448,23 +448,6 @@
                 option = -mlp64 ;
             }
         }
-        else
-        {
-            local arch = [ feature.get-values architecture : $(properties) ] ;
-            if $(arch) != arm
-            {
-                if $(model) = 32
-                {
-                    option = -m32 ;
-                }
-                else if $(model) = 64
-                {
-                    option = -m64 ;
-                }
-            }
-            # For darwin, the model can be 32_64. darwin.jam will handle that
-            # on its own.
-        }
         OPTIONS on $(targets) += $(option) ;
     }
 }
