--- festival/src/main/Makefile~	2008-10-28 16:24:22.000000000 +0000
+++ festival/src/main/Makefile	2013-07-08 09:28:20.075607941 +0100
@@ -50,13 +50,13 @@
 ETCDIR=$(TOP)/lib/etc/$(SYSTEM_TYPE)
 
 festival: festival_main.o $(LIBDEPS)
-	$(LINK_COMMAND) -o festival festival_main.o $(LIBS)
+	$(LINK_COMMAND) $(CFLAGS) -o festival festival_main.o $(LIBS)
 
 festival_client: festival_client.o $(REQUIRED_LIBDEPS)
-	$(LINK_COMMAND) -o festival_client festival_client.o $(LIBS)
+	$(LINK_COMMAND) $(CFLAGS) -o festival_client festival_client.o $(LIBS)
 
 $(ETCDIR)/audsp: $(ETCDIR)/.made audsp.o $(LIBDEPS)
-	$(LINK_COMMAND) -o $(ETCDIR)/audsp audsp.o $(LIBS)
+	$(LINK_COMMAND) $(CFLAGS) -o $(ETCDIR)/audsp audsp.o $(LIBS)
 
 # Can't just rely on the dir as it gets updated with new files
 # check for the data of a file created in etcdir
--- festival/examples/Makefile~	2008-11-23 21:07:52.000000000 +0000
+++ festival/examples/Makefile	2013-07-08 09:30:00.285424283 +0100
@@ -60,7 +60,7 @@
 	chmod +x $@
 
 festival_client: festival_client.o festival_client.h
-	$(LINK_COMMAND) -o festival_client festival_client.o $(LIBS)
+	$(LINK_COMMAND) $(CFLAGS) -o festival_client festival_client.o $(LIBS)
 
 festival_client.o: festival_client.c festival_client.h
 	$(CC_COMMAND) -DSTANDALONE festival_client.c -o festival_client.o
